### PR TITLE
8344336: SM cleanup of java.lang.System, Runtime, String, StackWalker

### PR DESCRIPTION
**Pull request by RogerRiggs**
Removed use of SecurityManager to check permissions.
Relaxed defensive copying of byte arrays (due to lack of SM)

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344336](https://bugs.openjdk.org/browse/JDK-8344336): SM cleanup of java.lang.System, Runtime, String, StackWalker (**Enhancement** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22168/head:pull/22168` \
`$ git checkout pull/22168`

Update a local copy of the PR: \
`$ git checkout pull/22168` \
`$ git pull https://git.openjdk.org/jdk.git pull/22168/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22168`

View PR using the GUI difftool: \
`$ git pr show -t 22168`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22168.diff">https://git.openjdk.org/jdk/pull/22168.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22168#issuecomment-2480043293)
</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/LiveStackFrame.java
Comment (eirbjo) [inline src/java.base/share/classes/java/lang/LiveStackFrame.java:1]: Seems you remove "@throws SecurityException" for the `getStackWalker` override actually throwing, but not for the delegating `getStackWalker` overrides?

File: src/java.base/share/classes/java/lang/String.java [hunk 1]
```diff
@@ -850,9 +840,8 @@ private static String newStringNoRepl1(byte[] src, Charset cs) {
     private static final char REPL = '\ufffd';
 
     // Trim the given byte array to the given length
-    @SuppressWarnings("removal")
-    private static byte[] safeTrim(byte[] ba, int len, boolean isTrusted) {
-        if (len == ba.length && (isTrusted || System.getSecurityManager() == null)) {
+    private static byte[] safeTrim(byte[] ba, int len) {
```
[the remark this answers is not in the document]
Comment (AlanBateman) [reply src/java.base/share/classes/java/lang/String.java:843]: Right, in other previously prepared changes, this was renamed to trim.

File: src/java.base/share/classes/java/lang/String.java [hunk 2]
```diff
@@ -685,12 +685,6 @@ private String(Charset charset, byte[] bytes, int offset, int length) {
             cd.onMalformedInput(CodingErrorAction.REPLACE)
                     .onUnmappableCharacter(CodingErrorAction.REPLACE);
             char[] ca = new char[en];
-            if (charset.getClass().getClassLoader0() != null &&
-                    System.getSecurityManager() != null) {
-                bytes = Arrays.copyOfRange(bytes, offset, offset + length);
-                offset = 0;
```
Comment (liach) [inline src/java.base/share/classes/java/lang/String.java:691]: Was this defensive copy particularly important to security managers but has no effect for other scenarios?

Comment (RogerRiggs) [reply src/java.base/share/classes/java/lang/String.java:691]: Application supplied Charsets are not any more or less trusted than any other class in the application.
So the copy is not (adequately) effective as a defensive measure here.

**Comment by AlanBateman** [comment]
What about System.codeSource? I assume it can be removed and Module.ensureNativeAccess changed to get the code source itself (no SM check needed now).

**Comment by RogerRiggs** [comment]
Blox'ed up this PR with duplicated Commits.  Will create a new PR.

**Pull request closed without merging**